device/src/main/java/se/sigmaconnectivity/blescanner/device/advertiser/BleAdvertiserImpl.kt

```diff
@@ -0,0 +1,59 @@
+package se.sigmaconnectivity.blescanner.device.advertiser
+
+import android.bluetooth.le.AdvertiseCallback
+import android.bluetooth.le.AdvertiseData
+import android.bluetooth.le.AdvertiseSettings
+import android.bluetooth.le.BluetoothLeAdvertiser
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+import se.sigmaconnectivity.blescanner.domain.ble.AdvertiserData
+import se.sigmaconnectivity.blescanner.domain.ble.BleAdvertiser
+import se.sigmaconnectivity.blescanner.domain.model.BLEFeatureState
+import se.sigmaconnectivity.blescanner.domain.model.StatusErrorType
+import timber.log.Timber
+
+abstract class BleAdvertiserImpl: BleAdvertiser {
```

Comment:
Moim zdaniem impl sugeruje, że to już ostateczna implementacja. Ja sugerowałbym BaseBleAdvertiser, a ewentualne Impl mogłbyby wtedy pojawić się u dzieci